packages/@aws-cdk/aws-rds/lib/database-secret.ts

```diff
@@ -55,5 +66,20 @@ export class DatabaseSecret extends secretsmanager.Secret {
         excludeCharacters: props.excludeCharacters ?? DEFAULT_PASSWORD_EXCLUDE_CHARS,
       },
     });
+
+    if (props.overrideLogicalId) {
+      const hash = crypto.createHash('md5');
+      hash.update(JSON.stringify({
+        path: this.node.path, // make it unique
+        // Use here the options that influence the password generation.
+        // If at some point we add other password customization options
+        // they sould be added here below (e.g. `passwordLength`).
+        excludeCharacters: props.excludeCharacters,
+      }));
+      const logicalId = `${this.node.id.replace(/[^A-Za-z0-9]/g, '')}${hash.digest('hex')}`;
```

Comment:
I feel like `id` is too short here. You'll wind up with `Secret<some-hash>` everywhere, which is too ambiguous. Can you use `uniqueId` here instead? (No need for the `replace` part then).